src/routes.js

```diff
@@ -6,19 +6,55 @@ import Classroom from './containers/Classroom';
 import Reference from './containers/Reference';
 import ReferenceExample from './containers/ReferenceExample';
 
-export default () => {
-  return (
-    <BrowserRouter>
-      <Switch>
-        <Route exact path='/' component={Ide} />
-        <Route exact path='/reference' component={Reference} />
-        <Route path='/course/:shortname' component={Guided} />
-        <Route path='/class/:classroom' component={Classroom} />
-        <Route path='/reference/:function' component={ReferenceExample} />
-        <Route path='/:id' component={Ide} />
-        <Redirect from="/lesson" exact to="/" />
-        <Redirect from="/course" exact to="/" />
-      </Switch>
-    </BrowserRouter>
-  );
-};
+class Routes extends React.Component {
```

Comment:
I think it's best if we leave the router as just a router. I would like the logic for this to be within just the view component itself.

Reply:
got it, should be done